packages/react/src/core/ContextMenu.tsx

```diff
@@ -128,3 +183,25 @@ function MenuItem({ icon, children, ...props }: { icon: string } & ComponentProp
     </Item>
   );
 }
+
+function Dialog({ onClose, children }: { onClose: () => void; children: ReactNode }) {
+  return (
+    <RadixDialog.Root open={true} onOpenChange={(open) => !open && onClose()}>
+      <RadixDialog.Portal>
+        <RadixDialog.Overlay className="fixed inset-0 opacity-50 bg-black" />
+
+        <RadixDialog.Content className="fixed top-50% left-50% transform-translate--50% w-90vw max-w-450px max-h-85vh rounded-xl text-tk-text-primary bg-tk-background-negative">
+          <div className="relative py-4 px-10">
+            <RadixDialog.Title className="text-6 mb-2">Error</RadixDialog.Title>
+
+            {children}
+
+            <RadixDialog.Close title="Close" className="absolute top-4 right-4 w-6 h-6">
+              <span aria-hidden className="i-ph-x block w-full h-full"></span>
+            </RadixDialog.Close>
+          </div>
+        </RadixDialog.Content>
+      </RadixDialog.Portal>
+    </RadixDialog.Root>
```

Comment:
@sulco, @Nemikolh, I wasn't really sure about what to put here and what kind of styles to use. Feel free to propose any changes here. Right now It's trying to follow TK's other styles, e.g. by using `bg-tk-background-negative`. 

In future (before next release) we will also show here error when user tries to create files that already exist on file system, e.g. creating `vite.config.ts` that's on template already. 

<img src="https://github.com/user-attachments/assets/2028586f-dd03-4ff0-9e56-98bdfef71b9e" width="400" />


<img src="https://github.com/user-attachments/assets/f8eec28b-7010-4438-97a2-b767e47ed5e2" width="400" />

Reply:
A few thoughts:
* We could probably follow what Svelte's tutorial is doing, aka:
  * Replace `Error` with `Cannot create this file` or `This action is not allowed`
  * Have an `Ok` button
  * Maybe let's use the app background color instead of the red which I find too strong
* I wonder if we should make that component overridable from the start, however a future PR is also fine

For reference this is what Svelte dialog looks like:
<img src="https://github.com/user-attachments/assets/e409dcd5-b3fa-4bd8-b784-0ccc7df45f55" width="500">

Reply:
Alright, now using following. What do you think?

And once I add the _"prevent creating files that exists on fs already, e.g. in templates"_ fix, we would show here `"This action is not allowed, File exists already"`.

![image](https://github.com/user-attachments/assets/b5a93450-9f00-4eac-8c09-c11fcb6b743c)

![image](https://github.com/user-attachments/assets/576591b5-36fb-4ed1-89e4-2bba2fdc963f)

Reply:
> I wonder if we should make that component overridable from the start, however a future PR is also fine

I agree this would be useful. But I'm not yet sure how this could be implemented, as this component is so deeply nested the `@tutorialkit/react` components. For override components, we need the logic to be in `@tutorialkit/astro` level. And previously override components could even be `.astro` files. 🤔

Reply:
I think we could provide the component via a React Context (or just do some form of dynamic linking). It would have the inconvenient that the default component is always included in the bundle but that's probably fine?